1.2/percentile_test.go

```diff
@@ -0,0 +1,59 @@
+package percentile
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestGeneratePercentile(t *testing.T) {
+	var tests = []struct {
+		fn percentile
+		p  uint
+	}{
+		{generatePercentile(.05), 15},
+		{generatePercentile(.3), 20},
+		{generatePercentile(.4), 20},
+		{generatePercentile(.5), 35},
+		{generatePercentile(1.0), 50},
+	}
+	data := []uint{15, 20, 35, 40, 50}
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("PASS"), func(t *testing.T) {
+			if output, err := test.fn(data); output != test.p || err != nil {
+				t.Errorf("Fail expected: %v", test.p)
+			}
+
+		})
+	}
+}
+
+func TestGeneratePercentileSecond(t *testing.T) {
+	var tests = []struct {
+		fn percentile
+		p  uint
+	}{
+		{generatePercentile(.25), 7},
+		{generatePercentile(.5), 8},
+		{generatePercentile(.75), 15},
+		{generatePercentile(1.0), 20},
+	}
+	data := []uint{3, 6, 7, 8, 8, 10, 13, 15, 16, 20}
```

Comment:
Can you add a test for the 0th percentile too, since your function explicitly handles that.